### PR TITLE
Custom auth segmentfetcher

### PR DESCRIPTION
**Pull request by chillaq**

### CODE REVIEW

File: client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java [hunk 1]
```diff
@@ -69,42 +66,27 @@ public SegmentChange fetch(String segmentName, long since, FetchOptions options)
...
-            if(options.cacheControlHeadersEnabled()) {
-                request.setHeader(CACHE_CONTROL_HEADER_NAME, CACHE_CONTROL_HEADER_VALUE);
-            }
-
-            response = _client.execute(request);
-
-            int statusCode = response.getCode();
-
-            if (_log.isDebugEnabled()) {
-                _log.debug(String.format("[%s] %s. Status code: %s", request.getMethod(), uri.toURL(), statusCode));
```
Comment (nmayorsplit) [inline client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java:83]: For me, we don't should remove this log. You can leave this log on the debug level to know what happens when we fetch segments.

**Review by nmayorsplit — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java (hunk 1 above)
Comment (chillaq) [reply client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java:83]: The log is in the SplitHttpClientImpl class

File: client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java [hunk 2]
```diff
@@ -37,16 +34,16 @@ public final class HttpSegmentChangeFetcher implements SegmentChangeFetcher {
     private static final String CACHE_CONTROL_HEADER_NAME = "Cache-Control";
     private static final String CACHE_CONTROL_HEADER_VALUE = "no-cache";
```
[the remark this answers is not in the document]
Comment (chillaq) [reply client/src/main/java/io/split/client/HttpSegmentChangeFetcher.java:35]: will do

**Review by nmayorsplit — APPROVED** [review]
LGTM

**Pull request merged**